packages/storage-ui/src/Contexts/StorageContext.tsx

```diff
@@ -340,17 +310,22 @@ const StorageProvider = ({ children }: StorageContextProps) => {
       return
     }
 
-    const id = uuidv4()
-    const uploadProgress: UploadProgress = {
-      id,
-      fileName: files[0].name,
-      complete: false,
-      error: false,
-      noOfFiles: files.length,
+    const cancelSource = axios.CancelToken.source()
+    const cancelToken = cancelSource.token
+
+    const toastParams: ToastParams = {
+      title: plural(files.length, {
+        one: `Uploading ${files.length} file`,
+        other: `Uploading ${files.length} files`
+      }) as string,
```

Comment:
plural returns a string, so that casting isn't needed.
```suggestion
      }),
```